### DISCUSSION TITLE
Clear selected values on RHF with MUI AutoComplete multiple

### DISCUSSION THREAD

**Discussion by shrikumarhegde** [react-hook-form/react-hook-form#11048] [Q&A]
Hi,
I am using the latest RHF for MUI autocomplete with multiple select options.
here is the demo link https://codesandbox.io/s/react-hook-form-with-autocomplete-7377js?file=/src/App.js

I want to clear the model dropdown value when the brand dropdown changes. 
I tried using setValue and resetField, it is clearing the data in hook form, but chips are still visible. I want to remove all selected chips when the model changes.

![image](https://github.com/react-hook-form/react-hook-form/assets/2070886/aa6a8543-9bb9-4ed7-bf3c-d5d90dc536ad)

**Comment by MikeyPenny** [comment]
I'm having the same problem here, I have an Autocomplete element and the reset method doesn't work for this type of fields, I tried the remove method in the useFieldArray hook without any result. 

Did you find a solution for this?

**Comment by chanthinh** [comment]
@shrikumarhegde,
You can check my codesandbox: https://codesandbox.io/p/sandbox/react-hook-form-with-autocomplete-forked-hcryd3?file=%2Fsrc%2FApp.js%3A10%2C17

**Comment by Shemrock3267** [comment]
Yep, same. Tried out this:
```js
const formData = form.getValues();
formData.forEach(fieldData => form.setValue(fieldData.field, null));
```

It does clear the value to null, but selected options can be seen inside Input

**Comment by ersanMobuy** [comment]
> Hi, I am using the latest RHF for MUI autocomplete with multiple select options. here is the demo link https://codesandbox.io/s/react-hook-form-with-autocomplete-7377js?file=/src/App.js
> 
> I want to clear the model dropdown value when the brand dropdown changes. I tried using setValue and resetField, it is clearing the data in hook form, but chips are still visible. I want to remove all selected chips when the model changes.
> 
> ![image](https://private-user-images.githubusercontent.com/2070886/275004248-aa6a8543-9bb9-4ed7-bf3c-d5d90dc536ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg4OTI5MDksIm5iZiI6MTcxODg5MjYwOSwicGF0aCI6Ii8yMDcwODg2LzI3NTAwNDI0OC1hYTZhODU0My05YmI5LTRlZDctYmYzYy1kNWQ5MGRjNTM2YWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjBUMTQxMDA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzdhNWU0YzBmZDZhODJhNGFlNzBiZGQyNmM1NzY3MWJjNGNjY2YwNjdhY2QxN2Q4MDVhMjY4OTM5M2EwNWJmZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dMaXIJbbZfMOhkxgKdzPbRLrq_xqq3vth8wnuSBihVg)

I have same problem, is tehere any solutions ?

**Comment by tatsmaki** [comment]
It is important to always set defaultValues in useForm 
```
 const form = useForm({
    defaultValues: {
      field1: "",
      field2: [],
    },
  });
```

https://codesandbox.io/p/sandbox/react-hook-form-with-autocomplete-forked-vglkxh

**Comment by sangecz** [comment]
only empty string works, not null nor undefined
you need to set your field in defaultValues also to empty string and then setValue('field', '') resets mui autocomplete
